src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -276,6 +276,26 @@ public static ImmutableArray<byte> ToImmutable(this MemoryStream stream)
             return builder.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Maps and flattens a subset of immutable array to another immutable array.
+        /// </summary>
+        /// <typeparam name="TItem">Type of the source array items</typeparam>
+        /// <typeparam name="TResult">Type of the transformed array items</typeparam>
+        /// <param name="array">The array to transform</param>
+        /// <param name="selector">A transform function to apply to each element.</param>
+        /// <returns>If the array's length is 0, this will return an empty immutable array.</returns>
+        public static ImmutableArray<TResult> SelectManyAsArray<TItem, TResult>(this ImmutableArray<TItem> array, Func<TItem, ImmutableArray<TResult>> selector)
```

Comment:
📝 There were 16 callers of the previous overload prior to this change. After this change, there are 3 callers of the previous overload, and 13 callers of the new overload.